### PR TITLE
Correctly handle the case when converting of value fails and return n…

### PR DESCRIPTION
**Pull request by normanmaurer**
…ull or default value.

Motivation:

Headers.get* methods should not throw an exception but return null or the default value if converting of the value fails.

Modifications:

- Correctly handle the case when ValueConverter throws an Exception.
- Add testcase.

Result:

Fixes [#7710].

### CODE REVIEW

File: codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java [hunk 1]
```diff
@@ -603,7 +603,11 @@ public T clear() {
     @Override
     public Boolean getBoolean(K name) {
         V v = get(name);
-        return v != null ? valueConverter.convertToBoolean(v) : null;
+        try {
+            return v != null ? valueConverter.convertToBoolean(v) : null;
+        } catch (Exception ignore) {
```
Comment (ejona86) [inline codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java]: nit: RuntimeException for clarity, although it is equivalent to what's here.

Comment (normanmaurer) [reply codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java]: @ejona86 We are talking about all the `catch` correct ? If so I think that makes sense @carl-mastrangelo @Scottmitch WDYT ?

Comment (ejona86) [reply codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java]: Yeah, I was talking about all of them

Comment (Scottmitch) [reply codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java]: `RuntimeException` sgtm

File: codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java [hunk 2]
```diff
@@ -603,7 +603,11 @@ public T clear() {
     @Override
     public Boolean getBoolean(K name) {
         V v = get(name);
-        return v != null ? valueConverter.convertToBoolean(v) : null;
+        try {
+            return v != null ? valueConverter.convertToBoolean(v) : null;
```
Comment (jkinkead) [inline codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java]: There is another bug here that I haven't filed: `convertToBoolean` has different behavior for `AsciiString` header values than for other `CharSequence` values.

`AsciiString` uses [this null-byte-check conversion](https://github.com/netty/netty/blob/4f982be91e6d61c7c7a9d8d63115e0ebe65b2774/common/src/main/java/io/netty/util/AsciiString.java#L1217-L1219), whereas other strings use [`Boolean.parseBoolean`](https://github.com/netty/netty/blob/4f982be91e6d61c7c7a9d8d63115e0ebe65b2774/codec/src/main/java/io/netty/handler/codec/CharSequenceValueConverter.java#L68-L73), which treats [only strings matching "true"](https://docs.oracle.com/javase/9/docs/api/java/lang/Boolean.html#parseBoolean-java.lang.String-) as truthy (case-insensitive).

Comment (normanmaurer) [reply codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java]: Please fill another bug or do a or to fix it

**Comment by normanmaurer** [comment]
@ejona86 @Scottmitch addressed PTAL again

File: codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java (hunk 2 above)
Comment (jkinkead) [reply codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java]: Done, issue #7717 . Thanks!

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (02b7507a622af049b7061fb2016ebb3b56b32111)

**Comment by Scottmitch** [comment]
lgtm

**Pull request closed without merging**